grpc-client-server/README.md

```diff
@@ -0,0 +1,188 @@
+[gRPC Core Concepts](https://grpc.io/docs/what-is-grpc/core-concepts/) for an overview of gRPC.
+todo stuff from primer
+
+## Get started with a HelloWorld gRPC example
+
+Start by building a simple gRPC ‘hello world’ server and cli client in golang. See [gRPC Quickstart](https://grpc.io/docs/languages/go/quickstart/) for detailed instructions.
+
+## Change HelloWorld to Prober
+
+Once you have the 'hello world' example from the gRPC Quickstart tutorial working, we will move on to implementing a prober.
+
+Rename `greeter_client` and `greeter_server` to `prober_client` and `prober_server`.   
+Change the `helloworld` subdirectory to be called `prober`,` and helloworld.proto` to `prober.proto`.
+Delete the genereted .go files in the same directory as `prober.proto`.
+
+Now update the the prober.proto file so that the main content section looks like this:
+
+```
+package prober;
+
+// The greeting service definition.
+service Prober {
+  // Sends a greeting
+  rpc DoProbes (ProbeRequest) returns (ProbeReply) {}
+}
+
+// The request message containing the user's name.
+message ProbeRequest {
+  string name = 1;
+}
+
+// The response message containing the greetings
+message ProbeReply {
+  string message = 1;
+}
+```
+
+Regenerate the generated protobuf code: 
+```
+protoc --go_out=. --go_opt=paths=source_relative \
+    --go-grpc_out=. --go-grpc_opt=paths=source_relative prober/prober.proto
+```
+
+Observe the new generated files:
+```
+ls prober
+prober.pb.go		prober.proto		prober_grpc.pb.go
+```
+
+Read through `prober_grpc.pb.go` - this is the interface you will use in your code. This is how gRPC works: a `proto` format 
+gets generated into language-specific code that you can use to interact with gRPCs. If you're working with multiple programming languages
+that need to interact through RPCs, you can do this by generating from the same protocol buffer definition with the language-specific
+tooling. 
+
+Now, update the client and server `main.go` files to use the changed package, service and message names.
+Get these running again as they did originally. The functionality will still be the same for now, just the hello world interaction.
+
+```
+go run prober_server/main.go
+2022/10/09 20:20:55 server listening at [::]:50051
+2022/10/09 20:21:24 Received: world
+````
+
+```
+go run prober_client/main.go 
+2022/10/09 20:21:24 Greeting: Hello world
+```
+
+While we haven't yet changed what the code does, we've gained some experience with the protocol buffer format, learned 
+how to generate Go code from protocol buffer definitions, and call that code from Go programs.
+
+Note: in a real production system you can't rename protocol buffers as we've done here: you would run into problems because your old clients can't talk to your new servers, and vice-versa. Since we gradually release code, we would have to support both services until all old clients were turned down.
```

Comment:
This feels like a bit of an educational anti-pattern: to get them to do something and they say "don't actually do this". Is there a way to structure this project to avoid it?